### PR TITLE
feat(modal-view-ios): handle iOS 13 dismiss modal gesture

### PR DESCRIPTION
**Pull request by ADjenkov**
- [iOS] Introduce `ShowModalOptions.cancelable` to enable/disable modal dismiss gesture.
- Deprecate `ShowModalOptions.android.cancelable` in favor of platform independent `ShowModalOptions.cancelable`

_Note1: `cancelable = true` by default on both platforms_
_Note2: more info about modality changes in iOS 13 [here](https://sarunw.com/posts/modality-changes-in-ios13/)_

Related to feature request https://github.com/NativeScript/NativeScript/issues/7941
Fix https://github.com/NativeScript/NativeScript/issues/7846

### CODE REVIEW

File: nativescript-core/ui/core/view/view.android.ts [hunk 1]
```diff
@@ -639,12 +639,21 @@ export class View extends ViewCommon {
...
         df.setArguments(args);
 
+        let cancelable = true;
+
+        if (options.android && (<any>options).android.cancelable !== undefined) {
+            cancelable = !!(<any>options).android.cancelable;
+            console.log("ShowModalOptions.android.cancelable is deprecated. Use ShowModalOptions.cancelable instead.");
+        }
+
+        cancelable = options.cancelable !== undefined ? options.cancelable : cancelable;
```
Comment (manoldonev) [inline nativescript-core/ui/core/view/view.android.ts]: For consistency maybe should be `!!options.cancelable`?

File: nativescript-core/ui/core/view/view.ios.ts [hunk 2]
```diff
@@ -422,17 +423,20 @@ export class View extends ViewCommon {
...
-                const view = parent.nativeViewProtected;
-                // Note: sourceView and sourceRect are needed to specify the anchor location for the popover.
-                // Note: sourceView should be the button triggering the modal. If it the Page the popover might appear "behind" the page content
-                popoverPresentationController.sourceView = view;
-                popoverPresentationController.sourceRect = CGRectMake(0, 0, view.frame.size.width, view.frame.size.height);
+                this._setupPopoverControllerDelegate(controller, parent);
             }
         }
 
+        const cancelable = options.cancelable !== undefined ? options.cancelable : true;
```
Comment (manoldonev) [inline nativescript-core/ui/core/view/view.ios.ts]: For consistency with Android maybe should be `!!options.cancelable`?

File: nativescript-core/ui/core/view/view.ios.ts [hunk 3]
```diff
@@ -422,17 +423,20 @@ export class View extends ViewCommon {
...
+
+        if (majorVersion >= 13 && cancelable) {
```
Comment (manoldonev) [inline nativescript-core/ui/core/view/view.ios.ts]: Better:

``` ts
if (majorVersion >= 13) {
    if (cancelable) {
        // Listen for dismiss modal callback.
        this._setupAdaptiveControllerDelegate(controller);
    } else {
        // Prevent users from dismissing the modal.
        (<any>controller).modalInPresentation = true;
    }
}
```

File: nativescript-core/ui/core/view/view.ios.ts [hunk 4]
```diff
@@ -422,17 +423,20 @@ export class View extends ViewCommon {
...
+            // Listen for dismiss modal callback.
+            this._setupAdaptiveControllerDelegate(controller);
+        } else if (majorVersion >= 13) {
+            // Prevent users from dismissing the modal.
+            (<any>controller).modalInPresentation = true;
+        }
```
Comment (manoldonev) [inline nativescript-core/ui/core/view/view.ios.ts]: Better:
``` ts
if (majorVersion >= 13) {
    if (cancelable) {
        // Listen for dismiss modal callback.
        this._setupAdaptiveControllerDelegate(controller);
    } else {
        // Prevent users from dismissing the modal.
        (<any>controller).modalInPresentation = true;
    }
}
```

**Comment by brandonfraune** [comment]
Can someone provide an ETA for when this work will be released?

**Comment by DimitarTachev** [comment]
@brandonfraune, It should be already available in the 6.3 release.

**Comment by brandonfraune** [comment]
@DimitarTachev Thanks, I must've missed that.

**Pull request merged**